File: src/test/resources/expected-output/java/model/TicketAttachmentV1Dto.java
package no.tensio.coreit.test.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

import java.time.LocalDateTime;
import java.util.UUID;

@Schema(name = "TicketAttachmentV1", description = "A media file or document attached to a ticket by reporter, case officer or other parties")
public record TicketAttachmentV1Dto (

    @Schema(description = "Unique ticket attachment identifier", format = "uuid")
    UUID id,

    @Schema(description = "Filename of attachment", required = true)
    @NotBlank
    String filename,

    @Schema(description = "Date and time of ticket attachment upload", required = true, format = "date-time")
    @NotNull
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    LocalDateTime uploadedTime,

    @Schema(description = "Description of attachment", required = true)
    @NotBlank
    String description,

    @Schema(description = "IANA media type descriptor for attachment content", required = true)
    @NotBlank
    String contentType,

    @Schema(description = "Size of attachment in number of bytes", required = true)
    @NotNull
    Integer contentLength
) {

}

File: src/test/resources/expected-output/kotlin/model/TicketAttachmentV1Dto.kt
package io.github.torand.test.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime
import java.util.UUID
import org.eclipse.microprofile.openapi.annotations.media.Schema

@Schema(name = "TicketAttachmentV1Dto", description="A media file or document attached to a ticket by reporter, case officer or other parties")
@JvmRecord
data class TicketAttachmentV1Dto (

    @field:Schema(description="Unique ticket attachment identifier", required = true, format = "uuid")
    @field:NotNull
    val id: UUID,

    @field:Schema(description="Filename of attachment", required = true)
    @field:NotBlank
    val filename: String,

    @field:Schema(description="Date and time of ticket attachment upload", required = true, format = "date-time")
    @field:NotNull
    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val uploadedTime: LocalDateTime,

    @field:Schema(description="Description of attachment", required = true)
    @field:NotBlank
    val description: String,

    @field:Schema(description="IANA media type descriptor for attachment content", required = true)
    @field:NotBlank
    val contentType: String,

    @field:Schema(description="Size of attachment in number of bytes", required = true)
    @field:NotNull
    val contentLength: Integer
)
